pages/index.tsx

```diff
@@ -1,33 +1,23 @@
 import type { NextPage } from "next";
 import React from "react";
-import { fetchSitSchedule } from "../lib/integration/sit";
-import { SitWeekSchedule } from "../types/integration/sit";
-import { ClassPopularityIndex } from "../types/rezervo";
-import { createClassPopularityIndex } from "../lib/popularity";
+import { fetchSitWeekSchedule } from "../lib/integration/sit";
+import { IntegrationPageProps } from "../types/rezervo";
 import Integration from "../components/Integration";
+import { getIntegrationPageStaticProps } from "../lib/integration/common";
+import { sitToRezervoWeekSchedule } from "../lib/integration/adapters";
 
-export async function getStaticProps() {
-    const initialCachedSchedules = await fetchSitSchedule([-1, 0, 1, 2, 3]);
-    const classPopularityIndex = await createClassPopularityIndex(initialCachedSchedules[-1]!);
-    const invalidationTimeInSeconds = 60 * 60;
-
-    return {
-        props: {
-            initialCachedSchedules,
-            classPopularityIndex,
-        },
-        revalidate: invalidationTimeInSeconds,
-    };
+export async function getStaticProps(): Promise<{
+    revalidate: number;
+    props: IntegrationPageProps;
+}> {
+    return await getIntegrationPageStaticProps(fetchSitWeekSchedule, sitToRezervoWeekSchedule);
 }
 
-const Index: NextPage<{
-    initialCachedSchedules: { [weekOffset: number]: SitWeekSchedule };
-    classPopularityIndex: ClassPopularityIndex;
-}> = ({ initialCachedSchedules, classPopularityIndex }) => {
+const Index: NextPage<IntegrationPageProps> = (props) => {
```

Comment:
Why no destructuring?

Reply:
Dunno why I changed it really 🤷🏻 Now it is destructured!